python/pluscodes.py

```diff
@@ -0,0 +1,515 @@
+            codeArea.longitudeLo + gridArea.longitudeLo,
+            codeArea.latitudeLo + gridArea.latitudeHi,
+            codeArea.longitudeLo + gridArea.longitudeHi,
+            codeArea.codeLength + gridArea.codeLength)
+
+"""
+ Recover the nearest matching code to a specified location.
+ Given a short Open Location Code of between four and seven characters,
+ this recovers the nearest matching full code to the specified location.
+ The number of characters that will be prepended to the short code, depends
+ on the length of the short code and whether it starts with the separator.
+ If it starts with the separator, four characters will be prepended. If it
+ does not, the characters that will be prepended to the short code, where S
+ is the supplied short code and R are the computed characters, are as
+ follows:
+ SSSS    -> RRRR.RRSSSS
+ SSSSS   -> RRRR.RRSSSSS
+ SSSSSS  -> RRRR.SSSSSS
+ SSSSSSS -> RRRR.SSSSSSS
+ Note that short codes with an odd number of characters will have their
+ last character decoded using the grid refinement algorithm.
+ Args:
+   shortCode: A valid short OLC character sequence.
+   referenceLatitude: The latitude (in signed decimal degrees) to use to
+       find the nearest matching full code.
+   referenceLongitude: The longitude (in signed decimal degrees) to use
+       to find the nearest matching full code.
+ Returns:
+   The nearest full Open Location Code to the reference location that matches
+   the short code. Note that the returned code may not have the same
+   computed characters as the reference location. This is because it returns
+   the nearest match, not necessarily the match within the same cell. If the
+   passed code was not a valid short code, but was a valid full code, it is
+   returned unchanged.
+"""
+def recoverNearest(shortcode, referenceLatitude, referenceLongitude):
+    if not isShort(shortcode):
+        raise ValueError('Passed short code is not valid - ' + str(shortcode))
+    # Ensure that latitude and longitude are valid.
+    referenceLatitude = clipLatitude(referenceLatitude)
+    referenceLongitude = normalizeLongitude(referenceLongitude)
+    # Clean up the passed code.
+    shortcode = shortcode.upper()
+    # Compute the number of digits we need to recover.
+    paddingLength = SEPARATOR_POSITION_ - shortcode.find(SEPARATOR_)
+    # The resolution (height and width) of the padded area in degrees.
+    resolution = pow(20, 2 - (paddingLength / 2))
+    # Distance from the center to an edge (in degrees).
+    areaToEdge = resolution / 2.0
+    # Now round down the reference latitude and longitude to the resolution.
+    roundedLatitude = math.floor(referenceLatitude / resolution) * resolution
+    roundedLongitude = math.floor(referenceLongitude / resolution) * resolution
+    # Use the reference location to pad the supplied short code and decode it.
+    codeArea = decode(encode(roundedLatitude, roundedLongitude)[0:paddingLength] + shortcode)
```

Comment:
Could you remove the roundedLatitude and roundedLongitude lines, and change the use in this line to referenceLatitude and referenceLongitude? (See issue #37.)